src/game/client/gui/gamewindow.cpp

```diff
@@ -665,4 +698,301 @@ void GameWindow::Win_Set_Edit_Data(GameWindowEditData *edit_data)
+    }
+
+    return 0;
+}
+
+int GameWindow::Win_Set_Input_Func(WindowCallbackFunc input)
+{
+    if (input != nullptr) {
+        m_input = input;
+    } else {
+        m_input = g_theWindowManager->Get_Default_Input();
+    }
+
+    return 0;
+}
+
+int GameWindow::Win_Set_System_Func(WindowCallbackFunc system)
+{
+    if (system != nullptr) {
+        m_system = system;
+    } else {
+        m_system = g_theWindowManager->Get_Default_System();
+    }
+
+    return 0;
+}
+
+int GameWindow::Win_Set_Callbacks(WindowCallbackFunc input, WindowDrawFunc draw, WindowTooltipFunc tooltip)
+{
+    Win_Set_Input_Func(input);
+    Win_Set_Draw_Func(draw);
+    Win_Set_Tooltip_Func(tooltip);
+    return 0;
+}
+
+int GameWindow::Win_Bring_To_Top()
+{
+    GameWindow *parent = Win_Get_Parent();
+
+    if (parent != nullptr) {
+        g_theWindowManager->Unlink_Child_Window(this);
+        g_theWindowManager->Add_Window_To_Parent(this, parent);
+    } else {
+        for (GameWindow *window = g_theWindowManager->Win_Get_Window_List(); window != nullptr; window = window->m_next) {
+            if (window == nullptr) {
+                return -3;
+            }
+        }
+
+        g_theWindowManager->Unlink_Window(this);
+        g_theWindowManager->Link_Window(this);
+    }
+
+    if (m_layout != nullptr) {
+        WindowLayout *layout = m_layout;
+        layout->Remove_Window(this);
+        layout->Add_Window(this);
+    }
+
+    return 0;
+}
+
+GameWindow *GameWindow::Find_First_Leaf()
+{
+    GameWindow *window = this;
+
+    while (window->m_parent != nullptr) {
+        window = window->m_parent;
+    }
+
+    while (window->m_child != nullptr) {
+        window = window->m_child;
+    }
+
+    return window;
+}
+
+GameWindow *GameWindow::Find_Last_Leaf()
+{
+    GameWindow *window = this;
+
+    while (window->m_parent != nullptr) {
+        window = window->m_parent;
+    }
+
+    while (window->m_child != nullptr) {
+        for (window = window->m_child; window->m_next != nullptr; window = window->m_next) {
+        }
+    }
+
+    return window;
+}
+
+GameWindow *GameWindow::Find_Next_Leaf()
+{
+    GameWindow *window = this;
+
+    if (m_prev != nullptr) {
+        GameWindow *prev = m_prev;
+
+        while (prev->m_child != nullptr && (prev->m_status & WIN_STATUS_TAB_STOP) == 0) {
+            for (prev = prev->m_child; prev->m_next != nullptr; prev = prev->m_next) {
+            }
+        }
+
+        return m_prev;
+    } else {
+        do {
+            if (window->m_parent == nullptr) {
+                if (window != nullptr) {
+                    return window->Find_Last_Leaf();
+                } else {
+                    return nullptr;
+                }
+            }
+
+            window = window->m_parent;
+        } while (window->m_parent == nullptr || window->m_prev == nullptr);
+
+        GameWindow *prev = m_prev;
+
+        while (prev->m_child != nullptr && (prev->m_status & WIN_STATUS_TAB_STOP) == 0) {
```

Comment:
This will crash, because `m_prev`, `prev` is null at this point.

Reply:
I had Find_Next_Leaf and Find_Prev_Leaf mixed up. I switched the names and verified that they are 100% correct to the original.

The function isn't even used anywhere that I have found (it doesn't even exist in game exe) so we could always just delete it and not implement it and make the problem go away but I would rather keep the "matches what I see in the original" version and leave it at that.